packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -241,3 +247,310 @@ export class Load extends Op {
+    hash.update(val);
+    const hashedOutput = Buffer.from(hash.hex(), 'hex');
+    var arrByte = Uint8Array.from(hashedOutput);
+    stack.push(arrByte);
+  }
+}
+
+// If A < B pushes '1' else '0'
+export class LessThan extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    if (a < b) {
+      stack.push(BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+    }
+  }
+}
+
+// If A > B pushes '1' else '0'
+export class GreaterThan extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    if (a > b) {
+      stack.push(BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+    }
+  }
+}
+
+// If A <= B pushes '1' else '0'
+export class LessThanEqualTo extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    if (a <= b) {
+      stack.push(BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+    }
+  }
+}
+
+// If A >= B pushes '1' else '0'
+export class GreaterThanEqualTo extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    if (a >= b) {
+      stack.push(BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+    }
+  }
+}
+
+// If A && B is true pushes '1' else '0'
+export class And extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    if (a && b) {
+      stack.push(BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+    }
+  }
+}
+
+// If A || B is true pushes '1' else '0'
+export class Or extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = this.assertBigInt(stack.pop());
+    const b = this.assertBigInt(stack.pop());
+    if (a || b) {
+      stack.push(BigInt('1'));
+    } else {
+      stack.push(BigInt('0'));
+    }
+  }
+}
+
+// If A == B pushes '1' else '0'
+export class EqualTo extends Op {
+  execute (stack: TEALStack): void {
+    this.assertStackLen(stack, 2);
+    const a = stack.pop();
+    const b = stack.pop();
+    if (typeof a === typeof b) {
+      if (typeof a === "bigint") {
+        if (a === b) {
+          stack.push(BigInt('1'));
+        } else {
+          stack.push(BigInt('0'));
+        }
+      } else {
+        if (compareArray(this.assertBytes(a), this.assertBytes(b))) {
+          stack.push(BigInt('1'));
+        } else {
+          stack.push(BigInt('0'));
+        }
+      }
+    } else {
+      throw new TealError(ERRORS.TEAL.INVALID_TYPE);
+    }
```

Comment:
Can't something like: `stack.push(BigInt(a === b));` work here?

Reply:
For `NotEqualTo` : `stack.push(BigInt(a !== b))`. Could you check?

Reply:
Please check how to compare two arrays.